client/modules/IDE/components/Header/MobileNav.jsx

```diff
@@ -0,0 +1,338 @@
+  const { t } = useTranslation();
+
+  const { pathname } = useLocation();
+  const editorLink = useSelector(selectSketchPath);
+
+  // TODO: remove the switch and use a props like mobileTitle <Nav layout=“dashboard” mobileTitle={t(‘Login’)} />
+  function resolveTitle() {
+    switch (pathname) {
+      case '/':
+        return project.name;
+      case '/login':
+        return t('LoginView.Login');
+      case '/signup':
+        return t('LoginView.SignUp');
+      case '/account':
+        return t('AccountView.Settings');
+      case '/p5/sketches':
+      case '/p5/collections':
+        return t('Nav.File.Examples');
+      case `/${user.username}/assets`:
+      case `/${user.username}/collections`:
+      case `/${user.username}/sketches`:
+        return 'My Stuff';
+      default:
+        return project.name;
+    }
+  }
+
+  const title = useMemo(resolveTitle, [project, pathname]);
+
+  const Logo = AsteriskIcon;
+  return (
+    <Nav>
+      <LogoContainer>
+        <Logo />
+      </LogoContainer>
+      <Title>
+        <h1>{title}</h1>
+        {project?.owner && title === project.name && (
+          <h5>by {project?.owner?.username}</h5>
+        )}
+      </Title>
+      <Options>
+        {user.authenticated ? (
+          <AccountMenu />
+        ) : (
+          <div>
+            <Link to="/login">
+              <IconButton icon={AccountIcon} />
+            </Link>
+          </div>
+        )}
+        {title === project.name ? (
+          <MoreMenu />
+        ) : (
+          <div>
+            <Link to={editorLink}>
+              <IconButton icon={EditorIcon} />
+            </Link>
+          </div>
+        )}
+      </Options>
+    </Nav>
+  );
+};
+
+const AccountMenu = () => {
+  const user = useSelector((state) => state.user);
+  const dispatch = useDispatch();
+
+  return (
+    <div>
+      <IconButton icon={AccountIcon} />
+      <ul>
+        <li className="user">{user.username}</li>
+        <li>
+          <Link to={`/${user.username}/sketches`}>
+            <button>My Stuff</button>
+          </Link>
+        </li>
+        <li>
+          <Link to="/account">
+            <button>Settings</button>
+          </Link>
+        </li>
+        <li>
+          <button onClick={() => dispatch(logoutUser())}>Log Out</button>
+        </li>
+      </ul>
+    </div>
+  );
+};
+
+const MoreMenu = () => {
+  // TODO: use selectRootFile selector
+  const rootFile = useSelector(
+    (state) => state.files.filter((file) => file.name === 'root')[0]
+  );
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const { newSketch, saveSketch } = useSketchActions();
+
+  const cmRef = useContext(CmControllerContext);
+
+  return (
+    <div>
+      <IconButton icon={MoreIcon} />
+      <ul>
+        <b>{t('Nav.File.Title')}</b>
+        <li>
+          <button onClick={newSketch}>{t('Nav.File.New')}</button>
+        </li>
+        <li>
+          <button onClick={() => saveSketch(cmRef.current)}>
```

Comment:
On the desktop nav we have some conditions to hide the "Save" button if the user is not logged in, so we should probably do that here.

Although the usability for clicking it when logged out is pretty decent, it opens a modal telling me to log in.

Reply:
I like that we're showing the modal to log in instead of hiding the save button when the user is not logged in, what do you think?

Reply:
Yeah I kind of like it too 🤷 Let's keep it for now but I don't like having different logic on mobile vs desktop.  When you get a chance maybe you can open an issue for discussion on how we should handle this.  Like whether we should show the "Save" option on desktop and have it open the modal.

We could also def make that modal nicer.  Emphasize the "Log In" action (maybe a big button?) and use a friendlier title than "Error".

Reply:
yeah I agree In the future we should make it nicer by adding a welcoming message with a login button to continue